pkg/controllers/state/cluster.go

```diff
@@ -412,6 +416,16 @@ func (c *Cluster) UpdatePodHealthyNodePoolScheduledTime(ctx context.Context, nod
 	}
 }
 
+// DeletePodHealthyNodePoolScheduledTime deletes podHealthyNodePoolScheduledTime
+// for pods that have pod errors
+func (c *Cluster) DeletePodHealthyNodePoolScheduledTime(podErrors map[*corev1.Pod]error, pods ...*corev1.Pod) {
```

Comment:
I feel like there's a way to collapse this code so that we just call `p.cluster.MarkPodSchedulingDecisions` and that automatically handles the marking and deleting for both pod schedulability and pod schedulability nodepool healthy.

Right now we have a bunch of functions, but I think we really only need one in reality:
- `MarkPodSchedulingDecisions`
- `MarkPodSchedulingDecisionsNodeRegistrationHealthy`
- `UpdatePodHealthyNodePoolScheduledTime`
- `UpdatePodHealthyNodePoolScheduledTime`
- `DeletePodHealthyNodePoolScheduledTime`

Reply:
Refactored the code.